### PR TITLE
Speedup for wavelets/final touchup/local contrast

### PR DESCRIPTION
**Pull request by heckflosse**
This is a draft for #6137

### CODE REVIEW

File: rtengine/ipwavelet.cc [hunk 1]
```diff
@@ -2216,67 +2221,61 @@ void ImProcFunctions::ip_wavelet(LabImage * lab, LabImage * dst, int kall, const
...
-                                float abs = pow(2.f * absciss, (1.f / SQR(sig)));
-                                absciss = 0.5f * abs;
-                            }
-                            float kc = wavguid.getVal(absciss) -1.f;
-
-                            if(kc < 0) {
-                                kc = -SQR(kc);//approximation to simulate sliders denoise
+                                absciss = amean * tempwav;
+                                /*
+                                if (sig == 2.f) { // for sig = 2.f we can use a faster calculation because the exponent in this case is 0.25
```
Comment (adamreichold) [inline rtengine/ipwavelet.cc:2265]: If we go for commented-out code, I would prefer to have a static assert here to check the value of `sig`.

Comment (adamreichold) [reply rtengine/ipwavelet.cc:2265]: Thinking about this some, we really should not do this: The comments are not compiled and will just rot away. Having the slow path will not affect the fast path at all as it should be removed by the optimizer. I think the the initial approach posted was superior.

Comment (Thanatomanic) [reply rtengine/ipwavelet.cc:2265]: I agree about comment-rot, we have a lot of that tbh. If anyone says/proves that the old, more verbose, way does not result in a significant slowdown, it should be kept. My argument would be that a lot of algorithms we use are not so well documented. Doing 'smart' optimizations in the code might lead to semantic-rot, which is worse imo. 
Edit: if there is a meaningful speed-up, I would actually prefer to keep the comments in. Development is currently done incrementally, tackling and improving little pieces here and there. Having some comments with old or alternative code might actually be a helpful reminder that something could still improve.

Comment (heckflosse) [reply rtengine/ipwavelet.cc:2265]: a static assert of a constexpr which is defined some lines before?

> If anyone says/proves that the old, more verbose, way does not result in a significant slowdown, it should be kept.

So I will keep it. The only downside is, that the file now is not cppcheck clean.

**Comment by heckflosse** [comment]
Any objections to merge this?

File: rtengine/ipwavelet.cc [hunk 2]
```diff
@@ -2216,67 +2223,57 @@ void ImProcFunctions::ip_wavelet(LabImage * lab, LabImage * dst, int kall, const
...
-                            }
-                            float kc = wavguid.getVal(absciss) -1.f;
-
-                            if(kc < 0) {
-                                kc = -SQR(kc);//approximation to simulate sliders denoise
+                                absciss = amean * tempwav;
+                                if (sig == 2.f) { // for sig = 2.f we can use a faster calculation because the exponent in this case is 0.25
+                                    absciss = 0.5f * std::sqrt(std::sqrt(absciss));
+                                } else { // original code for variable sig
+                                    absciss = 0.5f * pow_F(absciss, 1.f / k);
```
Comment (adamreichold) [inline rtengine/ipwavelet.cc:2268]: I may not be following the transformation correctly, but isn't there a factor two missing before the calls to `pow_F` and `sqrt` that was in the old code:
```c++
absciss = amean * std::fabs(tempwav);
float abs = pow(2.f * absciss, (1.f / SQR(sig)));
absciss = 0.5f * abs;
```

Comment (heckflosse) [reply rtengine/ipwavelet.cc:2268]: That factor of two went into an outside of loop calculation

Before it was
```
float amean = 0.5f / (tempmean);
```
Now it is
```
const float amean = 1.f / (tempmean);
```

Comment (adamreichold) [reply rtengine/ipwavelet.cc:2268]: Thanks! No objections then.

**Comment by Thanatomanic** [comment]
Go ahead for a merge! :+1:

**Pull request merged**